src/components/ha-md-dialog.ts

```diff
@@ -0,0 +1,96 @@
+import { MdDialog } from "@material/web/dialog/dialog";
+import { css } from "lit";
+import { customElement, property } from "lit/decorators";
+
+/**
+ * Based on the home assistant design: https://design.home-assistant.io/#components/ha-dialogs
+ *
+ */
+@customElement("ha-md-dialog")
+export class HaMdDialog extends MdDialog {
+  /**
+   * When true the dialog will not close when the user presses the esc key or press out of the dialog.
+   */
+  @property({ attribute: "disable-cancel-action", type: Boolean })
+  public disableCancelAction = false;
+
+  constructor() {
+    super();
+    this.open = true;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener("cancel", this._handleCancel);
+  }
+
+  _handleCancel(closeEvent: Event) {
```

Comment:
Pressing the escape key twice will still close the dialog, we should fix that

Reply:
This seems to be a chrome bug of the dialog implementation.
In Firefox it works. Can you test it in Safari please?
I will check if there is a way to fix it.

Reply:
Yeah, I see it behaves like this everywhere, and there is a proposal for changing the close behaviour. Let's keep it like this for now 👍